pkg/pillar/flextimer/flextimer.go

```diff
@@ -152,7 +152,12 @@ func flexTicker(config <-chan flexTickerConfig, tick chan<- time.Time) {
 		timer := time.NewTimer(d)
 		select {
 		case <-timer.C:
-			tick <- time.Now()
+			// this channel can not block, otherwise the config channel will block
+			// causing deadlock in zedagent
```

Comment:
Does anything care about missing a timer?
For example, if code does
1 update state X
2 timer fires, and zedagent starts processing state X
3 as part of that it updates state Y
4  timer fires, and we queue (channel length 1) the time.Now in this code, but it isn't processed until the event loop looks again
5 as part of processing X, zedagent updates state Z
6 timer fires, and this select loop doesn't send anything due to channel being full.
7. X processing done, back in main select loop and finds the time.Now from step 4
I guess the assumption is that zedagent would notice that it has work for both Y and Z in this case, hence the missing timer event doesn't matter.

Stated differently, the assumption is that the timer events are cumulative. 

Might make sense to add some more words to the comments to state this assumption that multiple timer events can be collapsed into one.